genie-web/src/main/java/com/netflix/genie/web/services/AgentRoutingService.java

```diff
@@ -63,4 +64,17 @@
      */
     void handleClientDisconnected(@NotBlank String jobId);
 
+    /**
```

Comment:
What about letting Spring do this, rather than having services explicitly register/unregister?